oasst-shared/oasst_shared/schemas/protocol.py

```diff
@@ -204,3 +204,51 @@ class PostRanking(Interaction):
     PostRating,
     PostRanking,
 ]
+
+
+class TextLabel(str, enum.Enum):
+    """A label for a piece of text."""
+
+    spam = "spam"
+    violence = "violence"
+    sexual_content = "sexual_content"
+    toxicity = "toxicity"
+    political_content = "political_content"
+    humor = "humor"
+    sarcasm = "sarcasm"
+    hate_speech = "hate_speech"
+    profanity = "profanity"
+    ad_hominem = "ad_hominem"
+    insult = "insult"
+    threat = "threat"
+    aggressive = "aggressive"
+    misleading = "misleading"
+    helpful = "helpful"
+    formal = "formal"
+    cringe = "cringe"
+    creative = "creative"
+    beautiful = "beautiful"
+    informative = "informative"
+    based = "based"
+    slang = "slang"
+
+
+class TextLabels(BaseModel):
+    """A set of labels for a piece of text."""
+
+    text: str
+    labels: dict[TextLabel, float]
+    post_id: str | None = None
+
+    @property
+    def has_post_id(self) -> bool:
+        """Whether this TextLabels has a post_id."""
+        return bool(self.post_id)
+
+    # check that each label value is between 0 and 1
```

Comment:
where did the enum __members__ check go? e.g. `str_value  in TextLabel.__members__`? didn't like it?

Reply:
it's now handled by the type system of defining `labels: dict[TextLabel, float]`

Reply:
ah, of course. ;-) nice.